lib/mongoid/shardable.rb

```diff
@@ -11,7 +11,7 @@ module Shardable
 
     included do
       cattr_accessor :shard_key_fields
-      self.shard_key_fields = []
+      self.shard_key_fields = {}
```

Comment:
This is an API change. Why was it made?

Reply:
I think if there is no one using shard_key, it may be better to change api. I think it is ugly to write like this
```ruby
self.shard_key_fields = []
self.shard_key_hash = {}
```

and there will be two method to declare a shard_key

```ruby
class User
  include Mongoid::Document
  include Mongoid::Timestamps
  
  shard_key :created_at
  shard_key_config created_at: "hashed"
end
```

`shard_key_hash` and `shard_key_config` may have a better name, but it will be confused to have two ways to declare shard_key